pymatgen/core/surface.py

```diff
@@ -869,25 +917,23 @@ def __init__(
         self.min_slab_size = min_slab_size
         self.in_unit_planes = in_unit_planes
         self.primitive = primitive
-        self._normal = normal
+        # self._normal = normal  # TODO (@DanielYang59): not used
+        self.reorient_lattice = reorient_lattice
+
         _a, _b, c = self.oriented_unit_cell.lattice.matrix
         self._proj_height = abs(np.dot(normal, c))
-        self.reorient_lattice = reorient_lattice
 
-    def get_slab(self, shift=0, tol: float = 0.1, energy=None):
-        """This method takes in shift value for the c lattice direction and
-        generates a slab based on the given shift. You should rarely use this
-        method. Instead, it is used by other generation algorithms to obtain
-        all slabs.
+    def _get_slab(self, shift: float=0, tol: float = 0.1, energy: float | None=None) -> Slab:
```

Comment:
Need discussion on this rename @janosh.

Though seemingly breaking, I guess this still should be renamed as a private method for two reasons:
- The original author of this code block explicitly suggested against using this method directly: https://github.com/materialsproject/pymatgen/blob/81dc677462d164182115903e28318cb5cf2ea804/pymatgen/core/surface.py#L877-L881
- There is another `get_slabs` method side by side, which is confusing: https://github.com/materialsproject/pymatgen/blob/81dc677462d164182115903e28318cb5cf2ea804/pymatgen/core/surface.py#L1038-L1052

Reply:
i suspect it's quite possible the `get_slab` method has seen external use despite it's doc string warning against it. so i don't think we can easily rename it at this point. i would keep it as is

Reply:
Is it really a warning against it or rather a suggestion about how the method should be used?

Reply:
yeah, warning was too strong a word. like you said, it's a suggestion

Reply:
> i suspect it's quite possible the `get_slab` method has seen external use despite it's doc string warning against it. so i don't think we can easily rename it at this point. i would keep it as is

Yes I totally agree with you on this point, as I noticed multiple external calls to this method in d3da8e29e89dba34568fccafdcd82e1a282bfe55.

> Is it really a warning against it or rather a suggestion about how the method should be used?

Yes I assume it's a suggestion too because `You should rarely use this method. Instead, it is used by other generation algorithms to obtain all slabs`.

But at the end of day, it's confusing to have two similar (in both name and function) methods side by side (`get_slab` and `get_slabs`), should we do something about this?

Reply:
i think it's not that confusing, especially with the `get_slab` doc string recommending to look at other methods

Reply:
Okay let's know it as is and I would revert this renaming.